### PR TITLE
enable buffered logs

### PR DESCRIPTION
**Pull request by SourceDoctor**
Option to enable buffering logs before writing to disc

### CODE REVIEW

File: templates/squid.conf.header.erb [hunk 1]
```diff
@@ -13,6 +13,7 @@ maximum_object_size_in_memory <%= @maximum_object_size_in_memory %>
 <% if @logformat -%>
 logformat                     <%= @logformat %>
 <% end -%>
+buffered_logs                 <%= (@buffered_logs == 'on' || @buffered_logs == true)?'on':'off' %>
```
Comment (alexjfisher) [inline templates/squid.conf.header.erb]: Probably best to wrap this in
```
unless @buffered_logs.nil?
```
and default the parameter to undef.

File: templates/squid.conf.header.erb [hunk 2]
```diff
@@ -13,7 +13,9 @@ maximum_object_size_in_memory <%= @maximum_object_size_in_memory %>
 <% if @logformat -%>
 logformat                     <%= @logformat %>
 <% end -%>
-buffered_logs                 <%= (@buffered_logs == 'on' || @buffered_logs == true)?'on':'off' %>
+<% unless @memory_cache_shared.nil? -%>
+buffered_logs                 <%= (@buffered_logs == true)?'on':'off' %>
```
Comment (alexjfisher) [inline templates/squid.conf.header.erb]: I guess the `== true` isn't needed?

File: templates/squid.conf.header.erb [hunk 3]
```diff
@@ -13,6 +13,9 @@ maximum_object_size_in_memory <%= @maximum_object_size_in_memory %>
 <% if @logformat -%>
 logformat                     <%= @logformat %>
 <% end -%>
+<% unless @memory_cache_shared.nil? -%>
```
Comment (alexjfisher) [inline templates/squid.conf.header.erb]: `memory_cache_shared`??

**Comment by SourceDoctor** [comment]
@alexjfisher 
everything done, something else to handle or ready to merge?

**Comment by alexjfisher** [comment]
@SourceDoctor Looks good.  Could you rebase please?

**Pull request merged**